src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -1090,22 +1094,21 @@ static enum e_op_result execute_stream_op(struct op_and_state* oas) {
       grpc_slice_buffer write_slice_buffer;
       grpc_slice slice;
       grpc_slice_buffer_init(&write_slice_buffer);
-      if (1 != stream_op->payload->send_message.send_message->Next(
-                   stream_op->payload->send_message.send_message->length(),
-                   nullptr)) {
-        /* Should never reach here */
-        GPR_ASSERT(false);
-      }
-      if (GRPC_ERROR_NONE !=
+      while (1 == stream_op->payload->send_message.send_message->Next(
```

Comment:
The definition of `ByteStream::Next` is that the return value indicates whether the next slice is immediately available (and can be pulled synchronously; if not, it will need to be pulled asynchronously. The current code assumed that the stream will contain exactly 1 slice, and that slice will always be immediately available and does not support otherwise.

So while it makes sense to fetch more than 1 slice from the byte stream, we need to decide to either continue support only the immediately available case (and assert if it is not the case), or to support async pull. For the former case, if the return value of `Next()` is false, we should assert "should never reach here". For the later case, we should have code for async read. The change here seems doing neither right now. Could you make some changes here? The former option would be much easier to implement and I will be fine of it for now given we're already doing that.

Reply:
done